### DISCUSSION TITLE
Minimum review periods

### DISCUSSION THREAD

**Discussion by ngerakines** [lexicon-community/lexicon#14] [Polls]
I want feedback on requiring a minimum amount of time between when a PR is created (or marked as ready if made as a draft) and when it can be merged after it has approval(s).

The community should be given ample time to consider requests and proposals. Having a minimum amount of time wouldn't prevent the PR author from starting to use their lexicon.

The minimum lifetime of a PR would be 2 weeks.

**Comment by snarfed** [comment]
I like the spirit behind this, but PRs vary widely. We obviously want a substantial review period for new lexicons, but we also obviously don't need it for eg typo fixes, and I don't know a concrete way to draw a line between the two, or whether there should be multiple categories, or something else altogether? Maybe only for the initial PR? 🤷

**Reply by ghost** [reply]
I think a process to gather consensus on closing a PR early should be drafted.

**Reply by rudyfraser** [reply]
Agree with this. Think there should just be consensus on what gets fast tracked vs what has the review window.
Even a smallish, non-backwards-compatible change to a widely used lexicon could be disruptive or confusing vs. large backwards compatible changes to a brand new lexicon

**Reply by ngerakines** [reply]
Agreed. Maybe this isn't a hard and fast rule, but rather an expectation set in the PR template that a technical streering committee member or collaborator can set a "Do Not Merge Before XYZ" date to PRs.

**Comment by tom-sherman** [comment]
I get the intention here but feel like it's much less black and white. As well as the type of change already mentioned there's the maturity of a lexicon itself to consider - for example I see no reason to require any delay on lexicons marked as experimental while ones used widely in production should probably have a longer than 2 week period.

I wonder if we first need to define a stability framework for lexicons that this kind of policy could hand off of? Something like https://tc39.es/process-document maybe?

**Reply by ngerakines** [reply]
Agreed and very reasonable. It's unfortunate that lexicons aren't very easily renamed or moved, because it'd be great to have some sort of testing indicating.

**Comment by bmann** [comment]
PR tooling can support N reviews needed can’t it? I’m hoping for a technical advisory committee sooner rather than later ;)

Agree with comments here that “it depends”. A lot of things have the potential to go from no usage to 100Ks very quickly which will make a lot of things challenging. 

Maybe I should start a new thread, but I think we’ll learn the most from actively collaborating. 

Is there appetite for a kind of “test harness” Lexicon?

Back in my Drupal days, we made a “Pants Module”, which always implemented all the new interfaces and features. We could play fast and loose with this Lexicon, and use it as a shared hack space.

**Reply by ngerakines** [reply]
> Is there appetite for a kind of “test harness” Lexicon?

That's a really interesting idea. :+1: to starting that thread and getting a larger lexicon testing discussion going. I love the idea.

**Comment by ngerakines** [comment]
Closing this as we've recently overhauled our governance repository.
